app/static/js/createEvents.js

```diff
@@ -173,49 +183,64 @@ $('#multipleOfferingSave').on('click', function() {
     }
   }
 
+  // Add past date validation
+  datePickerInputs.each(function(index, element) {
+    if (!allowPastStart && isDateInPast($(element).val())) {
+      $(element).addClass('border-red');
+      hasInvalidDates = true;
+    } else {
+      $(element).removeClass('border-red');
+    }
+  }); 
+
   if (isEmpty){
     let emptyFieldMessage = "Event name or date field is empty";
     displayNotification(emptyFieldMessage);
   }
-  else if(!hasValidTimes){
+  else if (!hasValidTimes) {
     let invalidTimeMessage = "Event end time must be after start time";
     displayNotification(invalidTimeMessage);
   }
-  else if (hasDuplicateListings){
+  else if (hasDuplicateListings) {
     let eventConflictMessage = "Event listings cannot have the same event name, date, and start time";
     displayNotification(eventConflictMessage);
   }
+  else if (hasInvalidDates) {
+    showModalFlashMessage ("Some events have dates in the past. Please correct them or enable 'Allow start date to be in the past'.", "danger");
+  }
   else {
-    saveOfferingsFromModal();
+    let offerings = [];
+    eventOfferings.each(function(index, element) {
+      offerings.push({
+        eventName: $(element).find('.multipleOfferingNameField').val(),
+        eventDate: $(element).find('.multipleOfferingDatePicker').val(),
+        startTime: $(element).find('.multipleOfferingStartTime').val(),
+        endTime: $(element).find('.multipleOfferingEndTime').val()
+      });
+    });
+
+    let offeringsJson = JSON.stringify(offerings);
+    $("#multipleOfferingData").val(offeringsJson);
+
     updateOfferingsTable();
     pendingmultipleEvents = [];
     $("#checkIsMultipleOffering").prop('checked', true);
-    // Remove the modal and overlay from the DOM
     $('#modalMultipleOffering').modal('hide');
-    $('.invalidFeedback').css('display', 'none');
-    $('#textNotifierPadding').removeClass('pt-5');
     msgFlash("Multiple time offering events saved!", "success");
   }
 });
 
+// Custom flash for modal: ensures message visibility when modal is open
+function showModalFlashMessage(message, type) {
```

Comment:
I noticed we have a displayNotification function that has a very similar functionality and is used in other locations. I checkout the branch and tried it it out and it looks to work the same is this function. Could we potentially use that instead so we don't have to be doing message flashes in multiple ways?